x-pack/solutions/security/plugins/security_solution/public/siem_migrations/rules/components/data_input_flyout/steps/rules/sub_steps/rules_file_upload/rules_file_upload.tsx

```diff
@@ -25,53 +31,86 @@ export interface RulesFileUploadProps {
 }
 export const RulesFileUpload = React.memo<RulesFileUploadProps>(
   ({ createMigration, apiError, isLoading, isCreated }) => {
-    const onFileParsed = useCallback(
-      (content: Array<SplunkRow<SplunkRulesResult>>) => {
-        const rules = content.map(formatRuleRow);
-        createMigration(rules);
-      },
-      [createMigration]
-    );
+    const [rulesToUpload, setRulesToUpload] = useState<CreateRuleMigrationRequestBody>([]);
+    const filePickerRef = useRef<EuiFilePickerClass>(null);
+
+    const createRules = useCallback(() => {
+      filePickerRef.current?.removeFiles();
+      createMigration(rulesToUpload);
+    }, [createMigration, rulesToUpload]);
+
+    const onFileParsed = useCallback((content: Array<SplunkRow<SplunkRulesResult>>) => {
+      const rules = content.map(formatRuleRow);
+      setRulesToUpload(rules);
+    }, []);
 
     const { parseFile, isParsing, error: fileError } = useParseFileInput(onFileParsed);
 
+    const onFileChange = useCallback(
+      (files: FileList | null) => {
+        setRulesToUpload([]);
+        parseFile(files);
+      },
+      [parseFile]
+    );
+
     const error = useMemo(() => {
       if (apiError) {
         return apiError;
       }
       return fileError;
     }, [apiError, fileError]);
 
+    const showLoader = isParsing || isLoading;
+    const isDisabled = showLoader || isCreated;
+    const isButtonDisabled = isDisabled || rulesToUpload.length === 0;
+
     return (
-      <EuiFormRow
-        helpText={
-          <EuiText color="danger" size="xs">
-            {error}
-          </EuiText>
-        }
-        isInvalid={error != null}
-        fullWidth
-      >
-        <EuiFilePicker
-          id="rulesFilePicker"
-          fullWidth
-          initialPromptText={
-            <>
-              <EuiText size="s" textAlign="center">
-                {i18n.RULES_DATA_INPUT_FILE_UPLOAD_PROMPT}
+      <EuiFlexGroup direction="column">
+        <EuiFlexItem>
+          <EuiFormRow
+            helpText={
+              <EuiText color="danger" size="xs">
+                {error}
               </EuiText>
-            </>
-          }
-          accept="application/json, application/x-ndjson"
-          onChange={parseFile}
-          display="large"
-          aria-label="Upload rules file"
-          isLoading={isParsing || isLoading}
-          disabled={isLoading || isCreated}
-          data-test-subj="rulesFilePicker"
-          data-loading={isParsing}
-        />
-      </EuiFormRow>
+            }
+            isInvalid={error != null}
+            fullWidth
+          >
+            <EuiFilePicker
+              id="rulesFilePicker"
+              ref={filePickerRef as React.Ref<Omit<EuiFilePickerProps, 'stylesMemoizer'>>}
+              fullWidth
+              initialPromptText={
+                <>
```

Comment:
Is the fragment necessary here?

Reply:
nope, this is a leftover from a previous version containing multiple elements. Changed in the 3 `EuiFilePicker` we are using here https://github.com/elastic/kibana/commit/90d829a67e1bedfafc4c30d3d5fe230b4f295af5